src/cargo/core/compiler/mod.rs

```diff
@@ -1865,12 +1865,15 @@ fn on_stderr_line_inner(
         return Ok(true);
     }
 
-    #[derive(serde::Deserialize)]
-    struct CompilerMessage {
-        level: String,
-    }
-    if let Ok(message) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
-        count_diagnostic(&message.level, options);
+    if let Ok(msg) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
```

Comment:
Instead of reusing a larger struct definition, should we just add an extra field `message`, and change `count_diagnostic` to accept three args `(level, message, options)` ?

Although having no data, my intuition tells me that skipping unused fields is generally more performant.


```json5
{
  "rendered": "error: expected one of `!` or `::`, found `<eof>`\n --> src/main.rs:3:1\n  |\n3 | xxx\n  | ^^^ expected one of `!` or `::`\n\n",
  "children": [], // may contain nested diagnostics
  "level": "error",
  "message": "expected one of `!` or `::`, found `<eof>`",
}
```

compares to

```json5
{
  "level": "error",
  "message": "expected one of `!` or `::`, found `<eof>`",
}
```

Reply:
Do you mean I should check the message in `count_diagnostic` instead of checking before passing the `(level, options` to `count_diagnostic`?

Reply:
Ignore the `count_diagnostic` part. I have no idea what I was talking about.

I meant that we may want to keep a separate `CompilerMessage` struct, since the current patch deserializes more fields than needed.

Reply:
fixed!